app/Model/EventReport.php

```diff
@@ -410,71 +406,86 @@ public function reArrangeReport(array $report)
     /**
      * getProxyMISPElements Extract MISP Elements from an event and make them accessible by their UUID
      *
-     * @param  array $user
-     * @param  int|string $eventid
+     * @param array $user
+     * @param int|string $eventid
      * @return array
+     * @throws Exception
      */
     public function getProxyMISPElements(array $user, $eventid)
     {
-        $event = $this->Event->fetchEvent($user, ['eventid' => $eventid]);
+        $options = [
+            'noSightings' => true,
+            'sgReferenceOnly' => true,
+            'noEventReports' => true,
+            'noShadowAttributes' => true,
+        ];
+
+        $event = $this->Event->fetchEvent($user, array_merge(['eventid' => $eventid], $options));
         if (empty($event)) {
             throw new NotFoundException(__('Invalid Event'));
         }
         $event = $event[0];
-        $parentEventId = $this->Event->fetchSimpleEventIds($user, ['conditions' => [
-            'uuid' => $event['Event']['extends_uuid']
-        ]]);
-        $completeEvent = $event;
-        if (!empty($parentEventId)) {
-            $parentEvent = $this->Event->fetchEvent($user, ['eventid' => $parentEventId, 'extended' => true]);
+
+        if (!empty($event['Event']['extends_uuid'])) {
+            $parentEvent = $this->Event->fetchEvent($user, array_merge(['event_uuid' => $event['Event']['extends_uuid']], $options));
             if (!empty($parentEvent)) {
-                $parentEvent = $parentEvent[0];
-                $completeEvent = $parentEvent;
+                $event = $parentEvent[0];
             }
         }
-        $attributes = Hash::combine($completeEvent, 'Attribute.{n}.uuid', 'Attribute.{n}');
-        $this->AttributeTag = ClassRegistry::init('AttributeTag');
+
         $allTagNames = Hash::combine($event['EventTag'], '{n}.Tag.name', '{n}.Tag');
-        $attributeTags = Hash::combine($this->AttributeTag->getAttributesTags($completeEvent['Attribute'], true), '{n}.name', '{n}');
-        $parentEventTags = !empty($parentEvent) ? Hash::combine($parentEvent['EventTag'], '{n}.Tag.name', '{n}.Tag') : [];
-        $allTagNames = array_merge($allTagNames, $attributeTags, $parentEventTags);
+
+        $attributes = [];
+        foreach ($event['Attribute'] as $attribute) {
+            unset($attribute['ShadowAttribute']);
+            $attributes[$attribute['uuid']] = $attribute;
+            foreach ($attribute['AttributeTag'] as $at) {
+                $allTagNames[$at['Tag']['name']] = $at['Tag'];
+            }
+            $this->Event->Attribute->removeGalaxyClusterTags($attribute);
```

Comment:
Should `removeGalaxyClusterTags` be moved up? It seems to do nothing useful.

Reply:
Nice catch, my mistake.